build.sbt

```diff
@@ -157,12 +159,19 @@ lazy val `jsoniter-scala-macros` = crossProject(JVMPlatform, JSPlatform)
   .settings(commonSettings)
   .settings(publishSettings)
   .settings(
-    crossScalaVersions := Seq("2.13.7", "2.12.15", "2.11.12"),
-    libraryDependencies ++= Seq(
-      "org.scala-lang" % "scala-compiler" % scalaVersion.value,
-      "org.scala-lang" % "scala-reflect" % scalaVersion.value,
-      "com.beachape" %%% "enumeratum" % "1.6.1" % Test,
-      "org.scalatest" %%% "scalatest" % "3.2.10" % Test
+    crossScalaVersions := Seq("3.1.0","2.13.7", "2.12.15", "2.11.12"),
```

Comment:
hmm.. would be surprised if the scala macros would build on  < 3.0 versions?

Reply:
currently the old code in 2.13.7 directory.  Probably it should be duplicated in 2.12.5 and 2.11.2. and then we can simplify 2.x versions.

Reply:
Probably, the old version of `JsonCodecMaker.scala` need to be moved into `scala-2` (not `scala-2.13` as proposed currently) to be accessible for 2.12 and 2.11 versions of cross-build too.

Reply:
@rssh That could be managed out by renaming `scala-2.13` from `jsoniter-scala-macros/shared/src/main` to `scala-2` and adding it to the list of unmanaged sources for the jsoniter-scala-macros module:
```
    Compile / unmanagedSourceDirectories ++=
      CrossVersion.partialVersion(scalaVersion.value) match {
        case Some((2, _)) => CrossType.Full.sharedSrcDir(baseDirectory.value, "main").toList.map(f => file(f.getPath + "-2"))
        case _  => Nil
      },
```

Reply:
applied